sdk/core/azure-core/azure/core/pipeline/policies/base.py

```diff
@@ -41,62 +41,101 @@
 
 class HTTPPolicy(ABC, Generic[HTTPRequestType, HTTPResponseType]): # type: ignore
     """An HTTP policy ABC.
+
+    Use with a synchronous pipeline.
+
+    :param next: Use to process the next policy in the pipeline. Set when pipeline is
+     instantiated and all policies chained.
+    :type next: HTTPPolicy or HTTPTransport
     """
     def __init__(self):
         self.next = None
 
     @abc.abstractmethod
     def send(self, request):
         # type: (PipelineRequest) -> PipelineResponse
-        """Mutate the request.
+        """Abstract send method for a synchronous pipeline. Mutates the request.
+
         Context content is dependent on the HttpTransport.
+
+        :param request: The pipeline request object
+        :type request: ~azure.core.pipeline.PipelineRequest
+        :return: The pipeline response object.
+        :rtype: ~azure.core.pipeline.PipelineResponse
         """
 
 class SansIOHTTPPolicy(Generic[HTTPRequestType, HTTPResponseType]):
     """Represents a sans I/O policy.
-    This policy can act before the I/O, and after the I/O.
-    Use this policy if the actual I/O in the middle is an implementation
-    detail.
-    Context is not available, since it's implementation dependent.
-    if a policy needs a context of the Sender, it can't be universal.
-    Example: setting a UserAgent does not need to be tied to
-    sync or async implementation or specific HTTP lib
+
+    If a policy just modifies or annotates the request based on the HTTP
+    specification, it's then a subclass of SansIOHTTPPolicy and will work
+    in either Pipeline or AsyncPipeline context. This is a simple abstract
+    class, that can act before the request is done, or after.
```

Comment:
As a user I prefer to not have to parse conditionals in documentation. Might suggest:

> SansIOHTTPPolicy is a base class for policies that only modify or mutate a request based on the HTTP specification, and do no depend on the specifics of any particular transport. SansIOHTTPPolicy subclasses will function in either a Pipeline or an AsyncPipeline, and can act either before the request is done, or after.